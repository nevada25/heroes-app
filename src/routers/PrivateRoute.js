import {Redirect, Route} from "react-router-dom";
import Protypes from 'prop-types';

export const PrivateRoute = ({isAuthenticated, component: Component, ...rest}) => {
    localStorage.setItem('lastPath', rest.location.pathname)
    return (
        <Route {...rest}
               component={(props) => (
                   (isAuthenticated)
                       ? (<Component {...props} />)
                       : (<Redirect to="/login"/>)
               )}/>
    );
}
PrivateRoute.propTypes = {
    isAuthenticated: Protypes.bool.isRequired,
    component: Protypes.func.isRequired,
}
