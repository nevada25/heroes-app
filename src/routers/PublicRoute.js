import {Redirect, Route} from "react-router-dom";
import Protypes from 'prop-types';

export const PublicRoute = ({isAuthenticated, component: Component, ...rest}) => {
    return (
        <Route {...rest}
               component={(props) => (
                   (!isAuthenticated)
                       ? (<Component {...props} />)
                       : (<Redirect to="/"/>)
               )}/>
    );
}
PublicRoute.propTypes = {
    isAuthenticated: Protypes.bool.isRequired,
    component: Protypes.func.isRequired,
}
