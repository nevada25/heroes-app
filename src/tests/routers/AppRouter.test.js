import {mount} from "enzyme";
import {AppRouter} from "../../routers/AppRouter";
import {AuthContext} from "../../auth/AuthContext";
import {MemoryRouter} from "react-router-dom";

describe('Pruebas en el <AppRouter/>', () => {
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    };
    test('Debe de mostrar login si no esta autenticado', () => {
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter/>
            </AuthContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar el componente marvel si esta autenticado', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user: {
                name: 'Kevin',
                logged: true
            }
        };
        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <AppRouter/>
                </MemoryRouter>
            </AuthContext.Provider>
        );
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.navbar').exists()).toBe(true);
    });

});
