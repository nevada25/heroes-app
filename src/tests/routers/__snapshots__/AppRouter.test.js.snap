// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <AppRouter/> Debe de mostrar login si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
